diff --git a/models/activities/user_heatmap.go b/models/activities/user_heatmap.go
index 332079952..ad502f880 100644
--- a/models/activities/user_heatmap.go
+++ b/models/activities/user_heatmap.go
@@ -39,6 +39,8 @@ func getUserHeatmapData(user *user_model.User, team *organization.Team, doer *us
 	groupBy := "created_unix / 900 * 900"
 	groupByName := "timestamp" // We need this extra case because mssql doesn't allow grouping by alias
 	switch {
+	case setting.Database.Cockroach:
+		groupBy = "created_unix // 900 * 900"
 	case setting.Database.Type.IsMySQL():
 		groupBy = "created_unix DIV 900 * 900"
 	case setting.Database.Type.IsMSSQL():
diff --git a/modules/setting/database.go b/modules/setting/database.go
index bb259f1fe..5aa7e7c73 100644
--- a/modules/setting/database.go
+++ b/modules/setting/database.go
@@ -28,6 +28,7 @@ var (
 	// Database holds the database settings
 	Database = struct {
 		Type              DatabaseType
+		Cockroach         bool
 		Host              string
 		Name              string
 		User              string
@@ -60,6 +61,7 @@ func LoadDBSetting() {
 func loadDBSetting(rootCfg ConfigProvider) {
 	sec := rootCfg.Section("database")
 	Database.Type = DatabaseType(sec.Key("DB_TYPE").String())
+	Database.Cockroach = Database.Type.IsPostgreSQL() && sec.Key("COCKROACH").MustBool(false)
 
 	Database.Host = sec.Key("HOST").String()
 	Database.Name = sec.Key("NAME").String()
